{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "2a100d2e-e2d4-4d02-97c5-036e3626b85c",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from Bio import Entrez\n",
    "Entrez.email = 'liaochen1988@gmail.com'"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d6ab62e6-3491-48d1-90e4-1d348372ac81",
   "metadata": {},
   "source": [
    "# Read strain information"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "23da3408-bb5d-4678-9a50-241d70711ba8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "100 sequences found.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Species</th>\n",
       "      <th>Original strain designation</th>\n",
       "      <th>DSM no.</th>\n",
       "      <th>Genbank (16S)</th>\n",
       "      <th>EzTaxon: best hit</th>\n",
       "      <th>EzTaxon: additional hits &gt;97 % sequence similarity</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genome accession</th>\n",
       "      <th>Growth atm.</th>\n",
       "      <th>Medium</th>\n",
       "      <th>Origin</th>\n",
       "      <th>Date isolation</th>\n",
       "      <th>Location</th>\n",
       "      <th>PI</th>\n",
       "      <th>Reference</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Acetatifactor muris</td>\n",
       "      <td>CT-m2</td>\n",
       "      <td>23669</td>\n",
       "      <td>HM989805</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Lachnospiraceae</td>\n",
       "      <td>EBI: PRJEB10572</td>\n",
       "      <td>anaerobe</td>\n",
       "      <td>see reference</td>\n",
       "      <td>caecal content; C57BL/6 obese mouse (high-calo...</td>\n",
       "      <td>Dec-2009</td>\n",
       "      <td>Freising, Germany</td>\n",
       "      <td>Clavel</td>\n",
       "      <td>Pfeiffer et al., 2012</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Actinomyces bowdenii</td>\n",
       "      <td>ERD15F</td>\n",
       "      <td>28589</td>\n",
       "      <td>KR364730</td>\n",
       "      <td>98.57% A. bowdenii M1327/96/1T (AJ234039)</td>\n",
       "      <td>none</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>Actinomycetaceae</td>\n",
       "      <td>EBI: PRJEB10572</td>\n",
       "      <td>aerobe</td>\n",
       "      <td>Blood agar</td>\n",
       "      <td>caecal content; TNFdeltaARE/+ C57BL/6 mouse</td>\n",
       "      <td>Oct-2008</td>\n",
       "      <td>Freising, Germany</td>\n",
       "      <td>Clavel</td>\n",
       "      <td>this paper</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>[Acutalibacter muris]</td>\n",
       "      <td>KB18</td>\n",
       "      <td>26090</td>\n",
       "      <td>KR364749</td>\n",
       "      <td>92.09% C. leptum DSM 753(T) (ABCB02000019)</td>\n",
       "      <td>none</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Ruminococcaceae</td>\n",
       "      <td>SRA: SRX1092360</td>\n",
       "      <td>anaerobe</td>\n",
       "      <td>BHI</td>\n",
       "      <td>feces; SPF TCRMOG92-106/I-As transgenic (RR) S...</td>\n",
       "      <td>Nov-2011</td>\n",
       "      <td>Munich, Germany</td>\n",
       "      <td>Stecher</td>\n",
       "      <td>Brugiroux et al., 2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Akkermansia muciniphila</td>\n",
       "      <td>YL44</td>\n",
       "      <td>26127</td>\n",
       "      <td>KR364731</td>\n",
       "      <td>99.86% Akkermansia muciniphila ATCC BAA-835(T)...</td>\n",
       "      <td>none</td>\n",
       "      <td>Verrucomicrobia</td>\n",
       "      <td>Akkermansiaceae</td>\n",
       "      <td>SRA: SRX1092354</td>\n",
       "      <td>anaerobe</td>\n",
       "      <td>WSB</td>\n",
       "      <td>C57BL/6J wildtype mouse</td>\n",
       "      <td>Oct-2009</td>\n",
       "      <td>Zurich, Switzerland</td>\n",
       "      <td>Stecher</td>\n",
       "      <td>Brugiroux et al., 2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Alistipes sp.</td>\n",
       "      <td>CC-5826-WT-bac</td>\n",
       "      <td>27924</td>\n",
       "      <td>KF809885</td>\n",
       "      <td>99.93% A. timonensis JC136(T) (CAEG01000022)</td>\n",
       "      <td>98.11% A. senegalensis JC50 (JF824804)</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Rikenellaceae</td>\n",
       "      <td>EBI: PRJEB10572</td>\n",
       "      <td>anaerobe</td>\n",
       "      <td>BAC</td>\n",
       "      <td>caecal content; wildtype C57BL/6 mouse</td>\n",
       "      <td>Dec-2012</td>\n",
       "      <td>Freising, Germany</td>\n",
       "      <td>Clavel</td>\n",
       "      <td>this paper</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   Species Original strain designation  DSM no. Genbank (16S)  \\\n",
       "0      Acetatifactor muris                       CT-m2    23669      HM989805   \n",
       "1     Actinomyces bowdenii                      ERD15F    28589      KR364730   \n",
       "2    [Acutalibacter muris]                        KB18    26090      KR364749   \n",
       "3  Akkermansia muciniphila                        YL44    26127      KR364731   \n",
       "4            Alistipes sp.              CC-5826-WT-bac    27924      KF809885   \n",
       "\n",
       "                                   EzTaxon: best hit  \\\n",
       "0                                                NaN   \n",
       "1          98.57% A. bowdenii M1327/96/1T (AJ234039)   \n",
       "2         92.09% C. leptum DSM 753(T) (ABCB02000019)   \n",
       "3  99.86% Akkermansia muciniphila ATCC BAA-835(T)...   \n",
       "4       99.93% A. timonensis JC136(T) (CAEG01000022)   \n",
       "\n",
       "  EzTaxon: additional hits >97 % sequence similarity           Phylum  \\\n",
       "0                                                NaN       Firmicutes   \n",
       "1                                               none   Actinobacteria   \n",
       "2                                               none       Firmicutes   \n",
       "3                                               none  Verrucomicrobia   \n",
       "4             98.11% A. senegalensis JC50 (JF824804)    Bacteroidetes   \n",
       "\n",
       "             Family Genome accession Growth atm.         Medium  \\\n",
       "0   Lachnospiraceae  EBI: PRJEB10572    anaerobe  see reference   \n",
       "1  Actinomycetaceae  EBI: PRJEB10572      aerobe     Blood agar   \n",
       "2   Ruminococcaceae  SRA: SRX1092360    anaerobe            BHI   \n",
       "3   Akkermansiaceae  SRA: SRX1092354    anaerobe            WSB   \n",
       "4     Rikenellaceae  EBI: PRJEB10572    anaerobe            BAC   \n",
       "\n",
       "                                              Origin Date isolation  \\\n",
       "0  caecal content; C57BL/6 obese mouse (high-calo...       Dec-2009   \n",
       "1        caecal content; TNFdeltaARE/+ C57BL/6 mouse       Oct-2008   \n",
       "2  feces; SPF TCRMOG92-106/I-As transgenic (RR) S...       Nov-2011   \n",
       "3                            C57BL/6J wildtype mouse       Oct-2009   \n",
       "4             caecal content; wildtype C57BL/6 mouse       Dec-2012   \n",
       "\n",
       "              Location       PI               Reference  \n",
       "0    Freising, Germany   Clavel   Pfeiffer et al., 2012  \n",
       "1    Freising, Germany   Clavel              this paper  \n",
       "2      Munich, Germany  Stecher  Brugiroux et al., 2016  \n",
       "3  Zurich, Switzerland  Stecher  Brugiroux et al., 2016  \n",
       "4    Freising, Germany   Clavel              this paper  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_excel(\"strain_info.xlsx\")\n",
    "df = df[df['Genbank (16S)'].notnull()]\n",
    "print(\"%d sequences found.\" % (len(df)))\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f49d3056-b492-4c6d-9bbd-b9bb2f4cab74",
   "metadata": {},
   "source": [
    "# Download sequences from NCBI and write to fasta file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "3d0e7e82-0f39-421d-b9a3-aa6b58ac9c4f",
   "metadata": {},
   "outputs": [],
   "source": [
    "out = open('miBC.fasta', 'w')\n",
    "for acc in df['Genbank (16S)']:\n",
    "    handle = Entrez.efetch(db=\"nucleotide\", id=acc, rettype=\"fasta\", retmode=\"text\")\n",
    "    record = handle.read()\n",
    "    record = record.split('\\n')[0] + '\\n' + ('').join(record.split('\\n')[1:]) + '\\n'\n",
    "    out.write(record)\n",
    "out.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3147eb58-6be3-476b-b431-71e83885264a",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
